library/lua/gui/widgets.lua

```diff
@@ -2,3338 +2,40 @@
-        math.max(0, math.min(df.global.world.map.y_count-1, anchor_pos.y)),
-        math.max(0, math.min(df.global.world.map.z_count-1, anchor_pos.z)))
-
-    return math.abs(mouse_pos.x - anchor_pos.x) + 1,
-        math.abs(mouse_pos.y - anchor_pos.y) + 1,
-        math.abs(mouse_pos.z - anchor_pos.z) + 1
-end
-
-function DimensionsTooltip:init()
-    ensure_key(self, 'frame').w = 17
-    self.frame.h = 4
-
-    self.label = Label{
-        frame={t=0},
-        auto_width=true,
-        text={
-            {
-                text=function()
-                    local anchor_pos = self.get_anchor_pos_fn()
-                    return ('%dx%dx%d'):format(get_cur_area_dims(anchor_pos))
-                end
-            }
-        },
-    }
-
-    self:addviews{
-        Panel{
-            -- set minimum size for tooltip frame so the DFHack frame badge fits
-            frame={t=0, l=0, w=7, h=2},
-        },
-        self.label,
-    }
-end
-
-function DimensionsTooltip:render(dc)
-    local x, y = dfhack.screen.getMousePos()
-    if not x or not self.get_anchor_pos_fn() then return end
-    local sw, sh = dfhack.screen.getWindowSize()
-    local frame_width = math.max(9, self.label:getTextWidth() + 2)
-    self.frame.l = math.min(x + self.display_offset.x, sw - frame_width)
-    self.frame.t = math.min(y + self.display_offset.y, sh - self.frame.h)
-    self:updateLayout()
-    DimensionsTooltip.super.render(self, dc)
-end
+Widget = require('gui.widgets.widget')
+Divider = require('gui.widgets.divider')
+Panel = require('gui.widgets.containers.panel')
+Window = require('gui.widgets.containers.window')
+ResizingPanel = require('gui.widgets.containers.resizing_panel')
+Pages = require('gui.widgets.containers.pages')
+EditField = require('gui.widgets.edit_field')
+HotkeyLabel = require('gui.widgets.labels.hotkey_label')
+Label = require('gui.widgets.labels.label')
+Scrollbar = require('gui.widgets.scrollbar')
+WrappedLabel = require('gui.widgets.labels.wrapped_label')
+TooltipLabel = require('gui.widgets.labels.tooltip_label')
+HelpButton = require('gui.widgets.buttons.help_button')
+ConfigureButton = require('gui.widgets.buttons.configure_button')
+BannerPanel = require('gui.widgets.containers.banner_panel')
+TextButton = require('gui.widgets.buttons.text_button')
+CycleHotkeyLabel = require('gui.widgets.labels.cycle_hotkey_label')
+ButtonGroup = require('gui.widgets.buttons.button_group')
+ToggleHotkeyLabel = require('gui.widgets.labels.toggle_hotkey_label')
+List = require('gui.widgets.list')
+FilteredList = require('gui.widgets.filtered_list')
+TabBar = require('gui.widgets.containers.tab_bar')
+RangeSlider = require('gui.widgets.range_slider')
+DimensionsTooltip = require('gui.widgets.dimensions_tooltip')
+
+Tab = TabBar.Tab
+makeButtonLabelText = Label.makeButtonLabelText
+parse_label_text = Label.parse_label_text
+render_text = Label.render_text
+check_text_keys = Label.check_text_keys
```

Comment:
are there any existing usages of these three functions (`parse_label_text`, `render_text`, and `check_text_keys`)? they aren't really intended for public use. they could probably stay as an internal API and not be exposed in `widgets`

Reply:
you are right.
removed from the public API.